src/ripple/app/main/NodeStoreScheduler.cpp

```diff
@@ -47,11 +47,14 @@ void NodeStoreScheduler::onChildrenStopped ()
 
 void NodeStoreScheduler::scheduleTask (NodeStore::Task& task)
 {
-    ++m_taskCount;
-    m_jobQueue->addJob (
-        jtWRITE,
-        "NodeObject::store",
-        [this, &task] (Job&) { doTask(task); });
+    if (! isStopping())
+    {
+        ++m_taskCount;
+        m_jobQueue->addJob (
+            jtWRITE,
+            "NodeObject::store",
+            [this, &task] (Job&) { doTask(task); });
+    }
```

Comment:
If you need to schedule a task to shut down, you should be able to do so. That's one of the reasons you were set to shut down before the job queue shuts down. Maybe the mistake is something that uses the node store (whatever did the wrote) reporting itself shut down before it's finished. Maybe the mistake is the node store reporting itself shut down before it finishes background writes.

I would not like to see this go in as is but instead maybe add asserts to track down the actual problem. Is something writing to the node store after the node store has reported shut down? Is the node store reporting itself shut down and then doing background writes?